daemons/attrd/attrd_cib.c

```diff
@@ -319,28 +382,33 @@ attrd_write_attribute(attribute_t *a, bool ignore_delay)
                  private_updates, pcmk__plural_s(private_updates),
                  a->id, pcmk__s(a->uuid, "n/a"), pcmk__s(a->set_id, "n/a"));
     }
-    if (cib_updates) {
-        crm_log_xml_trace(xml_top, __func__);
+    if (cib_updates > 0) {
+        char *id = NULL;
 
-        a->update = cib_internal_op(the_cib, PCMK__CIB_REQUEST_MODIFY, NULL,
-                                    XML_CIB_TAG_STATUS, xml_top, NULL, flags,
-                                    a->user);
+        // Call cib_internal_op() directly to pass a->user
+        a->update = cib_internal_op(the_cib, PCMK__CIB_REQUEST_COMMIT_TRANSACT,
+                                    NULL, NULL, the_cib->transaction, NULL,
+                                    cib_none, a->user);
```

Comment:
Not necessarily for this PR, but thinking out loud, I wonder if it would be worth a cib member/method for "set user" so the usual methods could be used everywhere